libtsuba/include/tsuba/ParquetWriter.h

```diff
@@ -20,7 +20,7 @@ class KATANA_EXPORT ParquetWriter {
     /// 2.0. In Arrow to Parquet version 1.0, nanosecond timestamps will get
     /// truncated to milliseconds.
     parquet::ParquetVersion::type parquet_version{
-        parquet::ParquetVersion::PARQUET_2_0};
+        parquet::ParquetVersion::PARQUET_2_6};
```

Comment:
Arrow 4.0, PARQUET_2_0 (e.g., support for ns timestamps) effectively means arrow 6.0 PARQUET_2_6. There is no PARQUET_2_0 is arrow 6.0.

Reply:
The associated [task](https://issues.apache.org/jira/browse/ARROW-13794) for 6.0.0 read:
```
ParquetVersion::type currently has two members: PARQUET_1_0 and PARQUET_2_0.
The latter enables a hodgepodge of features which were actually added
in Parquet format 2.4 and 2.6 (and others?).

To allow more fine-grained selection of the format version, we should deprecate
this value and add PARQUET_2_4 and PARQUET_2_6 values.
```

I lack any further info to understand how this will affect our backward compatibility. I would suspect that, since this only appears in the Writer, we should be able to read old data, but old versions may not be able to read newly-written data

Reply:
Thanks for flagging this case.  I cannot comment on compatibility but I will be on the lookout for anything related to this change.  @e-mcginnis can comment more if this would  impact anything about stored graphs.

Reply:
Thanks for the alert
Based on
>I would suspect that, since this only appears in the Writer, we should be able to read old data, but old versions may not be able to read newly-written data

that really sounds like we should be upreving the `storage_format_version`, since software without this PR will be unable to read rdgs output by software with this PR

The exact process to do this is still under active refinement, but the most important thing to do is bump https://github.com/KatanaGraph/katana/blob/1fbad29acbf94dca53cf04e8c59b76d14556cb6b/libtsuba/src/RDGPartHeader.h#L599 in this PR

Reply:
Just to be clear, PARQUET_2_0 (in libarrow 4.0) is PARQUET_2_6 (in libarrow 6.0):

https://github.com/apache/arrow/pull/11031/files#diff-fc3371fc0696bd56bf179db429eabc666d90eb854c934a09cbe58d511bd13546R43

If you thought PARQUET_2_0 was okay before, you should think the same about PARQUET_2_6.

Reply:
ah, I understand what you mean now @ddn0
though your use of "effectively" here 
> Arrow 4.0, PARQUET_2_0 (e.g., support for ns timestamps) effectively means arrow 6.0 PARQUET_2_6

worries me a bit

looking at that diff, it looks like we shouldn't see any difference between our usage of Arrow 4.0, PARQUET_2_0 and  Arrow 6.0 PARQUET_2_6

I think the proper test to ensure this is the case would be:
1) take an existing interesting rdg (ldbc_003 should be sufficient)
2) write out a copy of that rdg using the katana build by this PR 
3) load that rdg using katana master, load the original rdg, ensure they match

@danielmawhirter I can help you out with the specifics of how to go about this
Ensuring the original_rdg_propertygraph.Equals(new_rdg_property_graph) should be sufficient for (3) 

I'll get a test put together to do this more automatically in the future, but for now we will have to do it manually.

Reply:
"effectively" as in for "all intents and purposes".

Parquet only has two file format versions: 1 and 2.

Writers normalize their data types based on the minor version.

In arrow 6.0,

2.4: Writers use the new logical types supported by the parquet file format 2 but always truncate timestamps to microseconds
2.6: Writers do not truncate timestamps to microseconds.

In arrow 4.0, when the writer version is "2.0", writers use the new logical types supported by the parquet file format 2 and do not truncate timestamps to microseconds.

Compliant parquet file format 2.0 readers should have always supported both 2.4 and 2.6 style writers because they are just using the file format features supported by the underlying file. Though this is somewhat academic because I think the only implementation of parquet +2.0 readers or writers is libarrow.

You are welcome to make an explicit test for this though.

Reply:
thats convincing enough for me